app/javascript/src/utils/search.js

```diff
@@ -78,3 +78,17 @@ export function preloadSearchResults({
     )}`,
   );
 }
+
+export function fetchSearchAPIPromise(endpoint, dataHash) {
```

Comment:
I know this is still a draft, but for the final PR, we should definitely document how + when to use this helper with [JSDoc](https://jsdoc.app/) (which is what it looks like we've used somewhat sporadically in our JS files).